### PR TITLE
#167190422 implement user signup

### PR DESCRIPTION
**Pull request by obayomi96**
#### What does this PR do?
- Allows a user to register on the app
#### Description of Task to be completed?
- Write a Unit test
- Write integration test
- Write an end to end test
- Implement front end validation
- Implement the registration functionality

#### How should this be manually tested?
- Go to the home page
- Click the Signup button
- Enter valid details
- Submit the form

#### What are the relevant pivotal tracker stories?
#167190422

#### Screenshots (if appropriate)

The link below is the video to the end to end test

https://drive.google.com/open?id=1lZ3wZK0jTGa4Xat-MLIZUnhcWr6Unmdv

### CODE REVIEW

**Review by cvjude — APPROVED** [review]
LGTM

**Review by chuxmykel — APPROVED** [review]
Great Job Martins. I love the implementation for the confirm password field. Looks good to me

**Review by chiboycalix — APPROVED** [review]
LGTM

File: build-utils/webpack.common.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ module.exports = {
       '@Views': path.resolve(__dirname, '../', 'src/views/'),
       '@Utils': path.resolve(__dirname, '../', 'utils/'),
       '@Lib': path.resolve(__dirname, '../', 'src/lib/'),
+      '@Lib': path.resolve(__dirname, '../', 'src/lib/'),
```
Comment (topseySuave) [inline build-utils/webpack.common.js]: why is this added again?

Comment (obayomi96) [reply build-utils/webpack.common.js]: That was a mistake that wasn't meant to be there, apparently, I already added it before I rebased and didn't notice there was a duplicate. Would change it now.

**Review by Hadeneekeh — APPROVED** [review]
Nice work @obayomi96!!! I like the way you made use of the reusable components that @chuxmykel implemented.

LGTM

**Review by topseySuave — APPROVED** [review]
